SWLOR.Game.Server/Service/Communication.cs

```diff
@@ -77,8 +77,30 @@ public static void LoadHolonetSetting()
             SetLocalBool(player, "DISPLAY_HOLONET", dbPlayer.Settings.IsHolonetEnabled);
         }
 
+        /// <summary>
+        /// When a player focuses the chatbar, set a typing indicator on the player; when
+        /// unfocused, remove the indicator.
+        /// </summary>
+
+        [NWNEventHandler("mod_gui_event")]
+        public static void TypingIndicator()
+        {
+            var player = GetLastGuiEventPlayer();
+            var type = GetLastGuiEventType();
+            if (!GetIsPC(player)) return;
+
+            if(type == GuiEventType.ChatBarFocus)
+            {
+                var chatIndic = TagEffect(EffectVisualEffect(Core.NWScript.Enum.VisualEffect.VisualEffect.Vfx_Dur_Chat_Bubble, false, 0.5f), "typingindicator");
```

Comment:
Throw a using statement for the VisualEffect namespace on this file please. Easier to read without the fully qualified name.